reactor-netty-core/src/main/java/reactor/netty/udp/UdpResources.java

```diff
@@ -220,7 +225,34 @@ protected void _dispose() {
 	 * @return the Mono that represents the end of disposal
 	 */
 	protected Mono<Void> _disposeLater(Duration quietPeriod, Duration timeout) {
-		return defaultLoops.disposeLater(quietPeriod, timeout);
+		return Mono.when(_disposeResolverLater(), defaultLoops.disposeLater(quietPeriod, timeout));
+	}
+
+	AddressResolverGroup<?> getOrCreateDefaultResolver() {
+		AddressResolverGroup<?> resolverGroup = defaultResolver.get();
+		if (resolverGroup == null) {
+			AddressResolverGroup<?> newResolverGroup =
+					DEFAULT_NAME_RESOLVER_PROVIDER.newNameResolverGroup(defaultLoops, LoopResources.DEFAULT_NATIVE);
+			defaultResolver.compareAndSet(null, newResolverGroup);
```

Comment:
if the compareAndSet fails, is there any clean up to be done on the `newResolverGroup` variable?

Reply:
Will invoke `newResolverGroup.close()`